rotkehlchen/api/v1/resources.py

```diff
@@ -2293,11 +2298,6 @@ def put(self, file: Optional[Path], destination: Optional[Path], action: str) ->
                     status_code=HTTPStatus.BAD_REQUEST,
                 ))
             return self.rest_api.import_user_assets(path=file)
-        if destination is None:
```

Comment:
why remove this? Destination is optional according to the type

Reply:
I think I didn't communicate that well.

The endpoint basically uploads & downloads user added assets.
When `action=='upload`, `file` is needed but `destination` isn't.
When `action==download`, `destination` is optional because the behaviour is such that when it is provided, send the zip file to that path otherwise, send the zip file as a response to the user.

Reply:
All right I see. Thanks for the explanation.

But why did you have to change this in this PR. If I understand correctly this check was wrong from before too then?

I am actually a bit confused. Why do I see this diff in this commit?

When I check all files I don't see this having been there at all.

![2022-05-31_23-45](https://user-images.githubusercontent.com/1658405/171289685-5673cde6-9fc5-4ce5-8190-94c461481789.png)

Reply:
That's strange.

Yeah, it was there but after my refactor, mypy raised an error.
> error: Argument "path" to "import_user_assets" of "RestAPI" has incompatible type "Optional[Path]"; expected "Path"

Previously the endpoint looked like this.
https://github.com/prettyirrelevant/rotki/blob/develop/rotkehlchen/api/v1/resources.py#L2093

It can be changed to what's below since the validation already checks for a file during upload.
```py
if file is not None:
    return self.rest_api.import_user_assets(path=file)
```